upgrade to spring boot 3.0.2

rajadilipkolli opened the pull request:
- Upgrade to spring boot 3.0.2
- KafkaOperations has moved from ListenerFuture to CompletableFuture, changed code to adopt the same
- Switch to using Maven Wrapper
- Using OOTB cache provided by `actions/setup-java@v3.9.0` 
- Providing option to code using gitpod

rajadilipkolli committed:
upgrade to spring boot 3.0.1

timtebeek commented:
Love that you came along and fixed this! Very much appreciated! Did you make these changes by hand or did you use automation?

I'm traveling over the weekend, so will likely merge this Monday after a final check, but rest assured this will make it in!

rajadilipkolli committed:
Upgrade to spring boot 3.0.2

rajadilipkolli commented:
> Love that you came along and fixed this! Very much appreciated! Did you make these changes by hand or did you use automation?
> 
> I'm traveling over the weekend, so will likely merge this Monday after a final check, but rest assured this will make it in!

Hi @timtebeek , I have made changes manually as they are minor ones else I could have use `openrewrite` or `spring-boot-migrator` jar

switched to maven wrapper using `mvn wrapper:wrapper`

timtebeek committed:
Maven Wrapper 3.8.7

timtebeek commented on src/test/java/com/github/timtebeek/KafkaDeadLetterPublishingApplicationTests.java:
```diff
@@ -53,7 +54,7 @@ static void setProperties(DynamicPropertyRegistry registry) {
 	static void setup() {
 		// Create a test consumer that handles <String, String> records, listening to orders.DLT
-		// https://docs.spring.io/spring-kafka/docs/2.8.1/reference/html/#testing
+		// https://docs.spring.io/spring-kafka/docs/current/reference/html/#testing
```
Might seem odd at first, but I prefer to link to exact version numbers at stable URLs, rather than the moving target "current", which will mean something entirely different a year from now. It's happened before that links would break when using `current`.
```suggestion
		// https://docs.spring.io/spring-kafka/docs/3.0.x/reference/html/#testing
```

timtebeek committed:
Remove GitPod for now

timtebeek committed:
Pin documentation link

timtebeek commented:
Thanks again @rajadilipkolli !
